lib/install/validate/network.go

```diff
@@ -347,3 +373,78 @@ func (v *Validator) checkVDSMembership(ctx context.Context, network types.Manage
 	}
 	return nil
 }
+
+// suggestNetwork suggests all networks (only shows DPG on vC)
+func (v *Validator) suggestNetwork(flag string) {
+	defer trace.End(trace.Begin(flag))
+
+	log.Infof("Suggesting valid networks for %s", flag)
+
+	var validNets []string
+
+	nets, err := v.Session.Finder.NetworkList(v.Context, "*")
+	if err != nil {
+		log.Errorf("Unable to list networks: %s", err)
+		return
+	}
+
+	if len(nets) == 0 {
+		log.Info("No networks found")
+		return
+	}
+
+	for _, net := range nets {
+		switch o := net.(type) {
+		case *object.DistributedVirtualPortgroup: // for vC
+			if v.IsVC() && v.isNetworkNameValid(o.InventoryPath, flag) {
+				// Filter out DVS uplink
+				if !v.isDVSUplink(net.Reference()) {
+					validNets = append(validNets, v.inventoryPathToNetworkName(o.InventoryPath))
+				}
+			}
+		case *object.Network: // for ESX
+			if !v.IsVC() && v.isNetworkNameValid(o.InventoryPath, flag) {
+				validNets = append(validNets, v.inventoryPathToNetworkName(o.InventoryPath))
+			}
+		}
+	}
+
+	if len(validNets) == 0 {
+		log.Info("No valid networks found")
+		return
+	}
+
+	log.Infof("Suggested values for %s:", flag)
+	for _, n := range validNets {
+		log.Infof("  %q", n)
+	}
+}
+
+// isDVSUplink determines if the DVP is an uplink
+func (v *Validator) isDVSUplink(ref types.ManagedObjectReference) bool {
+	defer trace.End(trace.Begin(ref.Value))
+
+	var dvp mo.DistributedVirtualPortgroup
+
+	r := object.NewDistributedVirtualPortgroup(v.Session.Client.Client, ref)
+	if err := r.Properties(v.Context, r.Reference(), []string{"tag"}, &dvp); err != nil {
+		log.Errorf("Unable to check tags on %q: %s", ref, err)
+		return false
+	}
+	for _, t := range dvp.Tag {
+		if t.Key == "SYSTEM/DVS.UPLINKPG" {
```

Comment:
would suggest to check contains UPLINKPG or not. 
Another option is to check if there is one constant in govmomi for it, so avoid any VC internal message used here.